onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -339,4 +340,134 @@ Status LoadAttributeOrtFormat(const fbs::Attribute& fbs_attr,
+  tensor_name = fbs_tensor_name->str();
+
+  auto* tensor_dims = fbs_tensor.dims();
+  ORT_RETURN_IF_NOT(tensor_dims, "Checkpoint is invalid. Expected: Valid tensor dims. Actual: nullptr.");
+
+  auto tensor_data_type = fbs_tensor.data_type();
+  const DataTypeImpl* tensor_dtype = DataTypeImpl::TensorTypeFromONNXEnum(
+                                         static_cast<int>(tensor_data_type))
+                                         ->GetElementType();
+  auto dst_tensor = std::make_unique<onnxruntime::Tensor>(
+      tensor_dtype, TensorShape(InlinedVector<int64_t>{tensor_dims->begin(), tensor_dims->end()}), cpu_allocator);
+
+  // The tensor proto is used a dummy here. The actual data is stored in the raw_data field of the flatbuffer.
+  // The data is copied from the raw_data field to the dst_tensor.
+  ONNX_NAMESPACE::TensorProto unused_tensor_proto;
+  unused_tensor_proto.set_data_type(static_cast<int>(tensor_data_type));
+
+  switch (static_cast<int>(tensor_data_type)) {
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+      UNPACK_TESOR_WITH_TYPE(float);
+    case ONNX_NAMESPACE::TensorProto_DataType_BOOL:
+      UNPACK_TESOR_WITH_TYPE(bool);
+    case ONNX_NAMESPACE::TensorProto_DataType_DOUBLE:
+      UNPACK_TESOR_WITH_TYPE(double);
+    case ONNX_NAMESPACE::TensorProto_DataType_STRING:
+      UNPACK_TESOR_WITH_TYPE(std::string);
+    case ONNX_NAMESPACE::TensorProto_DataType_INT8:
+      UNPACK_TESOR_WITH_TYPE(int8_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT8:
+      UNPACK_TESOR_WITH_TYPE(uint8_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_INT16:
+      UNPACK_TESOR_WITH_TYPE(int16_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT16:
+      UNPACK_TESOR_WITH_TYPE(uint16_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_INT32:
+      UNPACK_TESOR_WITH_TYPE(int32_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT32:
+      UNPACK_TESOR_WITH_TYPE(uint32_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_INT64:
+      UNPACK_TESOR_WITH_TYPE(int64_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_UINT64:
+      UNPACK_TESOR_WITH_TYPE(uint64_t);
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
+    case ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E4M3FN:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E4M3FNUZ:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E5M2:
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT8E5M2FNUZ:
+    default:
+      return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED, "Cannot unpack tensor with type ",
+                             static_cast<int>(tensor_data_type));
+  }
+
+  ort_value.Init(dst_tensor.release(), DataTypeImpl::GetType<onnxruntime::Tensor>(),
+                 DataTypeImpl::GetType<onnxruntime::Tensor>()->GetDeleteFunc());
+
+  return Status::OK();
+}
+
+#undef UNPACK_TESOR_WITH_TYPE
+
+#endif
```

Comment:
general: for longer ifdef blocks it's helpful to show the original condition
```suggestion
#endif  // ENABLE_TRAINING_APIS
```